app/models/miq_worker.rb

```diff
@@ -152,7 +152,7 @@ def self.sync_workers
       _log.info("Workers are being synchronized: Current #: [#{current}], Desired #: [#{desired}]")
 
       if desired > current && enough_resource_to_start_worker?
-        (desired - current).times { result[:adds] << start_worker.pid }
+        (desired - current).times { result[:adds] << Proc.new { start_worker } }
       elsif desired < current
         w = w.to_a
         (current - desired).times do
```

Comment:
I'll see if I can do it easily in this PR otherwise, I might have to do it in a followup PR.  But yeah, thanks, it's a good idea.